main/models.py

```diff
@@ -78,3 +189,76 @@ def publish(self):
 
     def __str__(self):
         return self.name
+
+    def valid_location(self):
+        return (self.lng != 0 and self.lat != 0)
+
+    def as_geojson_feature(self):
+        return {
+            "type": "Feature",
+            "geometry": {
+                "type": "Point",
+                "coordinates": [float(self.lng), float(self.lat)]
+            },
+            "properties": {
+                "name": self.name,
+                "url": self.main_website_url,
+                "status": self.status,
+                "logo": self.logo_image_url
+            }
+        }
+
+
+class GocardlessPaymentManger(models.Manager):
+    # get_or_create that populates fields from json
+    # e.g. as received from gocardless webhook or payment creation api
+    # payload should be a dict, e.g. parsed from webhook json
+    def get_or_create_from_payload(self, payload):
+        try:
+            # create with required fields only
+            obj, created = super(GocardlessPaymentManger, self).get_or_create(
+                id = payload.id,
+                created_at = timezone.now(),
+                amount = payload.amount,
+                currency = payload.currency,
+                status = payload.status
+            )
+            # update everything else in the payload
+            for key, value in payload.__dict__.items():
+                if hasattr(obj, key):
+                    setattr(obj, key, value)
+            # don't forget the links
+            if hasattr(payload.links, 'mandate') and payload.links.mandate is not None:
+                obj.mandate_id = payload.links.mandate
+            if hasattr(payload.links, 'creditor') and payload.links.creditor is not None:
+                obj.creditor_id = payload.links.creditor
+            if hasattr(payload.links, 'payout') and payload.links.payout is not None:
+                obj.payout_id = payload.links.payout
+
+            obj.save()
+            return obj
+
+        except Exception as e:
+            logger.error("Error in get_or_create_from_payload - exception creating payment: " + repr(e), extra={'payload':payload})
+            return None
+
+
+class GocardlessPayment(models.Model):
```

Comment:
I think there should probably be a GocardlessMandate table which holds the mandate and creditor IDs, which don't change between payments. The payments table should key to that.